src/main/java/com/pnudev/librarysystem/mapper/UserMapper.java

```diff
@@ -1,13 +1,30 @@
 package com.pnudev.librarysystem.mapper;
 
+import com.pnudev.librarysystem.dto.CreateUserDTO;
+import com.pnudev.librarysystem.dto.UpdateUserDTO;
+import com.pnudev.librarysystem.dto.UserDTO;
 import com.pnudev.librarysystem.security.UserDetailsImpl;
 import com.pnudev.librarysystem.entity.User;
 import org.mapstruct.Mapper;
-
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
 
 @Mapper(componentModel = "spring")
```

Comment:
You may also use `uses` param to pass dependencies to be injected e.g. `PasswordEncoder`
Then it may be converted back to interface (I think)

Reply:
With `uses`,  `PasswordEncoder` is really injected, but it is also applied to all the fields of `String` type (🤷). 
Screen from generated implementation:

![image](https://user-images.githubusercontent.com/125756054/236698798-1202e874-1d60-4bf2-bfec-7e78a38d5c4b.png)

Reply:
Got it. Possible solutions are not really elegant so just leave it as is.